## Creamos una función para clasificar la habitación

In [5]:
def clasificar_habitacion(hab):
    # Convertimos a string para manejar los últimos 2 dígitos
    hab_str = str(hab)
    ultimos_dos = int(hab_str[-2:])
    primeros_dos = int(hab_str[:2])
    
    # Identificamos el piso por los primeros dos dígitos
    # Basado en el patrón: P(11), L(12), M(14)
    if primeros_dos >= 60: # Pisos 6, 7, 8, 9
        return "Piso Alto (Unido)"
    
    # Lógica para pisos bajos (P, L, M, 1-5)
    if ultimos_dos <= 24:
        return "Lado Norte"
    else:
        return "Lado Sur"

### Probamos la función aquí:

In [6]:
print(clasificar_habitacion(1038))
print(clasificar_habitacion(3040)) 
print(clasificar_habitacion(5025)) 

Lado Sur
Lado Sur
Lado Sur


## Algoritmo de Distribución

In [7]:
# 1. Entrada de datos interactiva

print("----- SISTEMA DE LOGÍSTICA DE REGALOS HOTEL -------")

# Pedimos las habitaciones separadas por comas (ej: 1010, 1260, 7001)

raw_habs = input("Ingresa los números de habitación separados por comas: ")

# Convertimos ese texto en una lista real de números

habitaciones_grupo = [int(h.strip()) for h in raw_habs.split(",")]

print("-"*51)
print("------------ REPORTE DE HABITACIONES: -------------")
print(f"\nTotal de habitaciones a repartir: {len(habitaciones_grupo)}")

num_bellboys = int(input("¿Cuántos Bellboys hay en turno?: ")) # Pedimos el número de bellboys

# 2. Listas vacías para organizar
norte, sur, altos = [], [], []

# 3. Clasificamos automáticamente
for hab in habitaciones_grupo:
    zona = clasificar_habitacion(hab)
    if zona == "Lado Norte":
        norte.append(hab)
    elif zona == "Lado Sur":
        sur.append(hab)
    else:
        altos.append(hab)

# Ordenamos cada lista para mejor organización
norte.sort()
sur.sort()
altos.sort()

ruta_maestra = norte + altos + sur # De esta manera se optimiza la repartición de habitaciones por bellboy
regalos_por_persona = len(ruta_maestra) // num_bellboys # los // aseguran un entero

# 4. Resultado final
print(f"\nLista de habitaciones clasificadas:")
print(f"\nLado Norte: {norte} Total: {len(norte)}")
print(f"Lado Sur: {sur} Total: {len(sur)}")
print(f"Pisos Altos: {altos} Total: {len(altos)}")
print("\n" + "-"*51)
print("-------------- HOJA DE ASIGNACIÓN -----------------")

for i in range(num_bellboys):
    inicio = i * regalos_por_persona # Calculamos el índice de inicio
    asignacion = ruta_maestra[inicio:] if i == num_bellboys - 1 else ruta_maestra[inicio:inicio + regalos_por_persona] # Este codigo asegura que el último bellboy tome cualquier resto
    print(f"BELLBOY {i+1}: {asignacion} Total: {len(asignacion)}")

----- SISTEMA DE LOGÍSTICA DE REGALOS HOTEL -------
---------------------------------------------------
------------ REPORTE DE HABITACIONES: -------------

Total de habitaciones a repartir: 37

Lista de habitaciones clasificadas:

Lado Norte: [1016, 1201, 1202, 1203, 1204, 2003, 2018, 2020, 3001, 4001, 4003, 4020, 5001] Total: 13
Lado Sur: [1026, 1029, 1170, 1255, 1256, 1257, 1260, 2038, 2040, 3040, 4025, 4030] Total: 12
Pisos Altos: [6003, 6008, 6010, 6023, 7001, 7020, 7021, 8001, 8016, 8021, 9005, 9018] Total: 12

---------------------------------------------------
-------------- HOJA DE ASIGNACIÓN -----------------
BELLBOY 1: [1016, 1201, 1202, 1203, 1204, 2003, 2018, 2020, 3001, 4001, 4003, 4020] Total: 12
BELLBOY 2: [5001, 6003, 6008, 6010, 6023, 7001, 7020, 7021, 8001, 8016, 8021, 9005] Total: 12
BELLBOY 3: [9018, 1026, 1029, 1170, 1255, 1256, 1257, 1260, 2038, 2040, 3040, 4025, 4030] Total: 13


### Se usaron estos numeros de habitación como ejemplo:

1202, 2003, 1201, 1203, 1204, 5001, 4020, 4003, 4001, 3001, 2018, 2020, 1016, 9018, 9005, 8021, 8016, 8001, 7021, 7001, 7020, 6003, 6008, 6010, 6023, 3040, 1255, 1256, 1257, 1260, 4025, 4030, 2040, 2038, 1029, 1026, 1170